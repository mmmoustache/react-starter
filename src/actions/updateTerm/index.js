const updateTerm = input => ({
  type: 'UPDATE_TERM',
  text: input || input,
});

export default updateTerm;
