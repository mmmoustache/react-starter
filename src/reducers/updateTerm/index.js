export default function (state = null, action) {
  switch (action.type) {
    case 'UPDATE_TERM':
      return action.text ? action.text : null;
      break;
    default:
      break;
  }
  return state;
}
