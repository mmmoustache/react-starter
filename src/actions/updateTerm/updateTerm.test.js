import updateTerm from '../updateTerm';

describe('Action Creator: updateTerm', () => {
  it('should match its snapshot', () => {
    const result = updateTerm();
    expect(result).toMatchSnapshot();
  })
});