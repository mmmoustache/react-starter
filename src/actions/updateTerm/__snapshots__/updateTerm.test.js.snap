// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Creator: updateTerm should match its snapshot 1`] = `
Object {
  "text": undefined,
  "type": "UPDATE_TERM",
}
`;
